tools/biotop.py

```diff
@@ -128,7 +128,19 @@ def signal_ignore(signal, frame):
     struct info_t info = {};
     info.major = req->rq_disk->major;
     info.minor = req->rq_disk->first_minor;
+/*
+ * The following deals with a kernel version change (in mainline 4.7, although
+ * it may be backported to earlier kernels) with how block request write flags
+ * are tested. We handle both pre- and post-change versions here. Please avoid
+ * kernel version tests like this as much as possible: they inflate the code,
+ * test, and maintenance burden.
+ */
+#ifdef REQ_WRITE
     info.type = req->cmd_flags & REQ_WRITE;
+#else
+    info.type = (req->cmd_flags >> REQ_OP_SHIFT) & REQ_OP_WRITE;
```

Comment:
i think it should (req->cmd_flags >> REQ_OP_SHIFT) == REQ_OP_WRITE
just like biosnoop